aria/data/datasets.py

```diff
@@ -74,12 +79,12 @@ def save(self, save_path: str):
     @classmethod
     def load(cls, load_path: str):
         """Loads dataset from JSON file."""
-        midi_dicts = []
-        with jsonlines.open(load_path) as reader:
-            for entry in reader:
-                midi_dicts.append(MidiDict.from_msg_dict(entry))
+        def _load():
+            with jsonlines.open(load_path) as reader:
+                for entry in reader:
+                    yield MidiDict.from_msg_dict(entry)
```

Comment:
This seems fine, but maybe include an option for when this functionality should be used e.g. stream=True or something.